ui/src/app/applications/components/application-resource-diff/application-resource-diff.tsx

```diff
@@ -16,42 +17,61 @@ export interface ApplicationComponentDiffProps {
 
 export const ApplicationResourceDiff = (props: ApplicationComponentDiffProps) => (
     <DataLoader load={() => services.viewPreferences.getPreferences()}>
-    {(pref) => {
-        let live = props.state.liveState;
-        if (pref.appDetails.hideDefaultedFields && live) {
-            live = removeDefaultedFields(props.state.targetState, live);
-        }
+        {(pref) => {
+            let live = props.state.liveState;
+            if (pref.appDetails.hideDefaultedFields && live) {
+                live = removeDefaultedFields(props.state.targetState, live);
+            }
 
-        const liveCopy = JSON.parse(JSON.stringify(live || {}));
-        let target: any = null;
-        if (props.state.targetState) {
-            target = props.state.diff ? jsonDiffPatch.patch(liveCopy, JSON.parse(props.state.diff)) : liveCopy;
-        }
+            const liveCopy = JSON.parse(JSON.stringify(live || {}));
+            let target: any = null;
+            if (props.state.targetState) {
+                target = props.state.diff ? jsonDiffPatch.patch(liveCopy, JSON.parse(props.state.diff)) : liveCopy;
+            }
 
-        return (
-            <div className='application-component-diff'>
-                <div className='application-component-diff__checkboxs'>
-                    <Checkbox id='inlineDiff' checked={pref.appDetails.inlineDiff}
-                            onChange={() => services.viewPreferences.updatePreferences({appDetails: {
-                                ...pref.appDetails, inlineDiff: !pref.appDetails.inlineDiff } })}/> <label htmlFor='inlineDiff'>
-                        Inline Diff
-                    </label>  <Checkbox id='hideDefaultedFields' checked={pref.appDetails.hideDefaultedFields}
-                            onChange={() => services.viewPreferences.updatePreferences({appDetails: {
-                                ...pref.appDetails, hideDefaultedFields: !pref.appDetails.hideDefaultedFields } })}/> <label htmlFor='hideDefaultedFields'>
-                        Hide default fields
-                    </label>
+            const a = live ? jsYaml.safeDump(live, {indent: 2}) : '';
+            const b = target ? jsYaml.safeDump(target, {indent: 2}) : '';
+            const context = pref.appDetails.compactDiff ? 2 : Number.MAX_SAFE_INTEGER;
+            const diffText = formatLines(diffLines(a, b), {context});
+            const files = parseDiff(diffText, {nearbySequences: 'zip'});
+            const viewType = pref.appDetails.inlineDiff ? 'unified' : 'split';
+            return (
+                <div className='application-component-diff'>
+                    <div className='application-component-diff__checkboxes'>
```

Comment:
Please don't forget to add left/right padding to have some extra space between checkboxes